types/node/v18/events.d.ts

```diff
@@ -380,10 +386,51 @@ declare module "events" {
              */
             signal?: AbortSignal | undefined;
         }
+
+        export interface EventEmitterReferencingAsyncResource extends AsyncResource {
+            readonly eventEmitter: EventEmitterAsyncResource;
+        }
+
+        export interface EventEmitterAsyncResourceOptions extends AsyncResourceOptions, EventEmitterOptions {
+            /**
+             * The type of async event.
+             * @default new.target.name
+             * @throws is required if `new.target === EventEmitterAsyncResource`
```

Comment:
I do not really understand what is the error throwed and when.  
Furthermore, I do not really understand why the `@throws` is put on an options interface member. 
I would put it on the `EventEmitterAsyncResource` constructor instead.

Reply:
I copied this from @KilianKilmister docs, the this implies that `name` is required when instantiating `EventEmitterAsyncResource` directly rather than extending it.

I'd agree that `@throws` is misplaced on an interface, I will re-word this.